R/dock_from_desc.R

```diff
@@ -218,26 +230,46 @@ dock_from_desc <- function(
       )
     )
 
+    nn <- sprintf(
+      "%s/%s",
+      nn_df$username,
+      nn_df$repo
+    )
+
+    repo_status <- lapply(nn, repo_get)
+    ind_private <- sapply(repo_status, function(x) x$visibility == "private") %|0|% FALSE
+    if (any(ind_private)) {
+      dock$ARG("GITHUB_PAT")
+      dock$RUN("GITHUB_PAT=$GITHUB_PAT")
+    }
+
+
     nn <- sprintf(
       "%s/%s@%s",
-      nn$username,
-      nn$repo,
-      nn$sha
+      nn_df$username,
+      nn_df$repo,
+      nn_df$sha
     )
 
 
     pong <- mapply(
-      function(dock, ver, nm) {
+      function(dock, ver, nm, i) {
+        fmt <- "Rscript -e 'remotes::install_github(\"%s\")'"
+        if (i)
+          fmt <- paste("GITHUB_PAT=$GITHUB_PAT", fmt)
```

Comment:
Can't we set `GITHUB_PAT`as a global environment variable instead when building the Docker using the `ARG` parameter ?  
You could set a `ARG GITHUB_PAT=default` that would be modified with `build-arg`.  
See for instance: https://stackoverflow.com/questions/39597925/how-do-i-set-environment-variables-during-the-build-in-docker

Indeed, this would not prevent us to configure {dockerfiler} for other sources than GitHub.   
At some point, we will use {attachment} to deal with other repositories sources (https://github.com/ThinkR-open/attachment/issues/56)

Reply:
Hi @statnmap, 
> You could set a ARG GITHUB_PAT=default that would be modified with build-arg.
See for instance: https://stackoverflow.com/questions/39597925/how-do-i-set-environment-variables-during-the-build-in-docker

That actually doesn't work. The `GITHUB_PAT` Environment variable has to be set each time R is invoked, otherwise it's not available to the R session when installing from private repositories. 
The drawback of the method we've provided in this PR is it actually exposes the GITHUB_PAT in the build log. If folks are using docker.io, most images are defaulted to public visibility after upload, thus exposing the GITHUB_PAT. 

I've been thinking through how best to ensure that the installation of remote private repos goes smoothly and the best way that I can think of seems to be to copy an _.Renviron_ file with the GITHUB_PAT variable set in it to the image while the installation of packages takes place. 

This could be accomplished by adding an argument that allows the user to copy the .Renviron to the image. If they opt to copy the _.Renviron_ then the `GITHUB_PAT` variable is just appended. If they do not opt to copy the _.Renviron_ file, then the file can be removed via `RUN rm .Renviron` at the end of the package installation process in the Dockerfile. 

What are your thoughts on this method of copying the _.Renviron_ with the GITHUB_PAT to the image either permanently or temporarily based on user preference?